src/frontend/deployment_details.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	DetailsAreNotLoaded = 0
+	DetailsAreLoading   = 1
+	DetailsAreLoaded    = 2
+)
+
+var deploymentDetailsMap = make(map[string]string)
+var detailsLoadingStatus = DetailsAreNotLoaded
+var detailsLoadingStatusMutex sync.Mutex // Ensures only 1 HTTP request loads deployment details.
+
+func loadDeploymentDetails(httpRequest *http.Request) map[string]string {
+	var metaServerClient = metadata.NewClient(&http.Client{})
+
+	// The use of httpRequest here lets us to see HTTP request info in the logs.
+	var log = httpRequest.Context().Value(ctxKeyLog{}).(logrus.FieldLogger)
```

Comment:
I think, ideally, this `loadDeploymentDetails` function would not know about the `httpRequest` object.
So, in the future, we could pass in the `log` object instead of the `httpRequest` object into the function.
But such an improvement would be outside the scope of this pull-request.